src/Utilities/TimerManager.cpp

```diff
@@ -74,6 +75,40 @@ TIMER* TimerManager<TIMER>::createTimer(const std::string& myname, timer_levels
   return t;
 }
 
+template<class TIMER>
+void TimerManager<TIMER>::push_timer(TIMER* t)
+{
+  // current_timer() can be nullptr when the stack was empty.
+  if (t == current_timer())
+  {
+    std::cerr << "Timer " << t->get_name()
+              << " instance is already at the top of the stack. "
+                 "start() is being called again. This often happens when stop() is not paired properly with start(). "
+              << "ScopedTimer uses RAII and managers timer start/stop much safer." << std::endl;
```

Comment:
Possible change: "ScopedTimer uses RAII and manages timer start/stop more safely."